creator-node/src/routes/files.js

```diff
@@ -192,13 +192,34 @@ module.exports = function (app) {
       if (req.query.filename) {
         res.setHeader('Content-Disposition', contentDisposition(req.query.filename))
       }
+
+      // Determine if cat one byte of ipfs CID takes over 500ms
+      await ipfsSingleByteCat(CID, req, 500)
+
+      // Stream file from ipfs if cat one byte takes under 500ms
       await new Promise((resolve, reject) => {
         req.app.get('ipfsAPI').catReadableStream(CID)
           .on('data', streamData => { res.write(streamData) })
           .on('end', () => { res.end(); resolve() })
           .on('error', e => { reject(e) })
       })
     } catch (e) {
+      // ipfsCatSingleByte took over 500ms, try streaming from file system
+      // If file cannot be found on disk nor ipfs, throw error
+      if (!fs.existsSync(queryResults.storagePath)) {
+        return sendResponse(req, res, errorResponseServerError('File could not be found on disk.'))
+      }
+
+      // Stream file from file system if cat one byte takes over 500ms or errs out
+      const fileStream = fs.createReadStream(queryResults.storagePath)
+
+      await new Promise((resolve, reject) => {
+        fileStream
+          .on('open', () => fileStream.pipe(res))
+          .on('end', () => { res.end(); resolve() })
+          .on('error', e => { reject(e) })
+      })
+
       return sendResponse(req, res, errorResponseServerError(e.message))
```

Comment:
unless i'm misunderstanding, it seems this 500 errorResponse is unreachable? optional exits are: line 210 returns error, line 214 fails (and throws out), line 219 returns success, or line 220 rejects and throws out. also seems like in this catch statement we are susceptible to errors that are uncaught. thoughts?

Reply:
yeah ^this is right. should probably a different function that tries to stream and subsequently returns 500?

Reply:
oh good catch! 

@SidSethi actually, upon success (line 219), it will send the errorResponse and the file (which is not what we want -- we would want the file only).  upon rejection (line 220) the system throw an error, but will not hit line 223 (thus not send the errorResponse).

@hareeshnagaraj can do that! would be nice to have abstracted helper methods to help with complexity. 

with both of your comments, i will refactor as necessary :) thanks team